vignettes/workflows.Rmd

```diff
@@ -0,0 +1,210 @@
+
+```yaml
+remote:
+  default:
+    driver: orderlyweb::orderlyweb_remote
+    args:
+      host: orderly.example.com
+      token: $GITHUB_TOKEN
+```
+
+Once this is done, both Alice and Bob can run reports *on* that server, for example
+
+```r
+orderly::orderly_run_remote("report-a", remote = "default")
+```
+
+**ADD A GIF**
+
+this will print to screen the log, just as if it was running locally, though it is in fact running on a fresh R session remotely.  After being run, this report is available to everyone and can be pulled as above
+
+```r
+orderly::orderly_pull_archive("report-a", remote = "default")
+```
+
+## Using staging environments
+
+This can be extended further with the use of "staging environments".  This extends the above (two) scenario so that we have multiple remote orderly archives.  These might be shared file archives, using `orderly::orderly_remote_path` or server hosted archives using `orderlyweb::orderlyweb_remote` or even a combination.
+
+We designate one of these remotes to be our production environment and the others to be staging.  In our work we have used the name pairs `production/science` and `real/testing`; the names used do not matter.
+
+The `orderly_config.yml` configuration might then look like:
+
+```yaml
+remote:
+  staging:
+    driver: orderlyweb::orderlyweb_remote
+    args:
+      host: orderly.example.com
+      port: 11443
+      token: $GITHUB_TOKEN
+  production:
+    driver: orderlyweb::orderlyweb_remote
+    args:
+      host: orderly.example.com
+      token: $GITHUB_TOKEN
+```
+
+Now, as part of the pull request review procedure, the reviewer will run the report on the staging environment, which ensures that it will likely work without error on the production environment.
+
+```r
+orderly::orderly_run_remote("report-a", remote = "staging", ref = "origin/myfeature")
+```
+
+where `myfeature` is the name of the branch.  This report can now be inspected (a URL will be generated which may be emailed around), and after being merged, the report can be run on the production server with
+
+```r
+orderly::orderly_run_remote("report-a", remote = "production")
+```
+
+We would recommend that on the production server only `master` be used.  This can be enforced by adding `master_only: true` to the configuration as:
+
+```
+remote:
+  production:
+    driver: orderlyweb::orderlyweb_remote
+    master_only: true
+    args:
+      ...
+```
+
+## Dependency resolution
+
+As soon as there are multiple archives, resolving dependencies becomes more challenging.  If Alice has two reports that depend on each other, she might want to use the most recent versions on her machine, or the most recent versions *as seen by one of her remotes* as that is where the report will eventually be run.
+
+To allow this, when running `orderly::orderly_run`, Alice can provide the argument `remote`, indicating which of her remotes she would like to use for dependency resolution.
+
+**GIVE A CONCRETE EXAMPLE HERE, WITH DIAGRAM**
```

Comment:
Yep - I think this is where it's easy to feel the slight plunge of "euuu, this is getting messy to think about"